index.js

```diff
@@ -212,12 +212,25 @@ class Request {
             }
 
             if (res.status >= 400) {
+                body = body || {};
+                if (Buffer.isBuffer(body)) {
+                    body = body.toString();
+                }
+                if (body.constructor !== Object) {
```

Comment:
Can body be an instance of `HTTPError`?

Reply:
Ah indeed. Changed the code to check for strings explicitly.